tests/test_tokenization.py

```diff
@@ -133,6 +133,64 @@ def fetch_from_2(self, query, parameters={}, literal_binds=True):
         query, parameters = self.compileQuery(query, parameters=parameters, literal_binds=literal_binds)
         return self.executor2.execute_prepared_statement(query, parameters)
 
+    def compile_execute_prepare(self, prepared_name, data={}):
+        if len(data) == 0:
+            prepare_query_sql = "execute {}".format(prepared_name)
+        else:
+            values_str = ''
+            for x in data.values():
+                if isinstance(x, int):
+                    values_str += '{}, '.format(str(x))
+                if x is None:
+                    values_str += 'null, '
+                if isinstance(x, str):
+                    values_str += '\'{}\', '.format(str(sa.text(x)))
+                if isinstance(x, bytes):
+                    if len(x) == 0:
+                        values_str += '\'\', '
+                    else:
+                        values_str += '\'\\x{}\', '.format(x.hex())
+
+            prepare_query_sql = "execute {} ({})".format(prepared_name, values_str.removesuffix(', '))
+        return prepare_query_sql
+
+    def compile_prepare(self, prepared_name, query, data_types={}, literal_binds=True):
+        query, _ = self.compileQuery(query, parameters=data_types, literal_binds=literal_binds)
+
+        if len(data_types) != 0:
+            prepare_query_sql = "prepare {} ({}) as {}".format(prepared_name, ", ".join(data_types.values()), query)
+        else:
+            prepare_query_sql = "prepare {} as {}".format(prepared_name, query)
+        return prepare_query_sql
+
+    def prepare_1(self, prepared_name, query, data_types={}, literal_binds=True):
```

Comment:
hm... lets break previous convention and acccept engine as parameter to not produce a lot of functions `x_1` and `x_2`? it will help to re-use them for other flows too